zebra-consensus/src/checkpoint/list.rs

```diff
@@ -55,41 +55,57 @@ impl ParameterCheckpoint for Network {
     fn genesis_hash(&self) -> zebra_chain::block::Hash {
         match self {
             // zcash-cli getblockhash 0
-            Network::Mainnet => "00040fe8ec8471911baa1db1266ea15dd06b4a8a5c453883c000b031973dce08",
-            // zcash-cli -testnet getblockhash 0
-            // TODO: Add a `genesis_hash` field to `testnet::Parameters` and return it here (#8366)
-            Network::Testnet(_params) => {
-                "05a60a92d99d85997cce3b87616c089f6124d7342af37106edc76126334a2c38"
-            }
+            Network::Mainnet => "00040fe8ec8471911baa1db1266ea15dd06b4a8a5c453883c000b031973dce08"
+                .parse()
+                .expect("hard-coded hash parses"),
+            // See `zebra_chain::parameters::network::testnet` for more details.
+            Network::Testnet(params) => params.genesis_hash(),
         }
-        .parse()
-        .expect("hard-coded hash parses")
     }
 
     fn checkpoint_list(&self) -> CheckpointList {
-        // parse calls CheckpointList::from_list
-        // TODO:
-        // - Add a `genesis_hash` field to `testnet::Parameters` and return it here (#8366)
-        // - Try to disable checkpoints entirely for regtest and custom testnets
-        let checkpoint_list: CheckpointList = match self {
-            Network::Mainnet => MAINNET_CHECKPOINTS
-                .parse()
-                .expect("Hard-coded Mainnet checkpoint list parses and validates"),
-            Network::Testnet(_params) => TESTNET_CHECKPOINTS
-                .parse()
-                .expect("Hard-coded Testnet checkpoint list parses and validates"),
+        let checkpoints_for_network = match self {
+            Network::Mainnet => MAINNET_CHECKPOINTS,
+            Network::Testnet(params) if !params.is_regtest() => TESTNET_CHECKPOINTS,
+            // Returns only the genesis checkpoint for Regtest
+            Network::Testnet(_params) => {
+                // It's okay to skip checking the hard-coded genesis checkpoint is the genesis checkpoint
+                return CheckpointList::from_list([(block::Height(0), self.genesis_hash())])
+                    .expect("hard-coded checkpoint list parses and validates");
```

Comment:
This should be moved to the match statement below when it's using any configured Testnet so that it's general to when the genesis hash from the `TESTNET_CHECKPOINTS` doesn't match the first checkpoint on Regtest or a configured Testnet.